### PR TITLE
Check for empty container roles set.

### PR DESCRIPTION
**Pull request by peichman-umd**
**JIRA Ticket**: https://jira.duraspace.org/browse/FCREPO-2961

# What does this Pull Request do?
Check for empty container roles set.

# What's new?
If there are no container roles, skip trying to call `addAll()` on the Shiro principal set, since an empty set causes Shiro to throw an exception. Updated the test authentication filter to have the user "noroles" be authenticated but have no container roles, for use in integration tests.

# How should this be tested?
The usual `mvn clean verify` should exercise the included unit and integration tests.

# Additional Notes:
None.

# Interested parties
@whikloj, @dbernstein, @fcrepo4/committers

### CODE REVIEW

File: fcrepo-auth-common/src/test/java/org/fcrepo/auth/common/ServletContainerAuthenticatingRealmTest.java [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+ */
+public class ServletContainerAuthenticatingRealmTest {
+
+    @Test
+    public void testDoGetAuthenticationInfo() {
+        final String username = "foo";
+        final String roleName = "fedoraUser";
+
+        final ServletContainerAuthenticatingRealm realm = new ServletContainerAuthenticatingRealm();
+        final Set<String> roles = new HashSet<String>();
```
Comment (awoods) [inline fcrepo-auth-common/src/test/java/org/fcrepo/auth/common/ServletContainerAuthenticatingRealmTest.java]: Minor: Use `<>` operator:
```
final Set<String> roles = new HashSet<>();
```

File: fcrepo-auth-common/src/test/java/org/fcrepo/auth/common/ServletContainerAuthenticatingRealmTest.java [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+        assertEquals(roleName, info.getPrincipals().byType(ContainerRolesPrincipal.class).toArray(
+                new ContainerRolesPrincipal[1])[0].getName());
+    }
+
+    @Test
+    public void testDoGetAuthenticationInfoWithNoRoles() {
+        final String username = "foo";
+
+        final ServletContainerAuthenticatingRealm realm = new ServletContainerAuthenticatingRealm();
+        final ContainerAuthToken token = new ContainerAuthToken(username, new HashSet<String>());
```
Comment (awoods) [inline fcrepo-auth-common/src/test/java/org/fcrepo/auth/common/ServletContainerAuthenticatingRealmTest.java]: Minor: Use `<>` operator:
```
final ContainerAuthToken token = new ContainerAuthToken(username, new HashSet<>());
```

File: fcrepo-auth-common/src/test/java/org/fcrepo/auth/integration/ServletContainerAuthenticatingRealmIT.java [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+        final String basic = "Basic " + encCreds;
+        method.setHeader("Authorization", basic);
+    }
+
+    @Test
+    public void testUserWithoutRoles() throws IOException {
+        // make sure this doesn't cause Shiro to explode
+        final HttpGet request = new HttpGet(serverAddress);
+        setAuth(request, "noroles");
+        execute(request);
```
Comment (awoods) [inline fcrepo-auth-common/src/test/java/org/fcrepo/auth/integration/ServletContainerAuthenticatingRealmIT.java]: Maybe check for a success status?

**Comment by whikloj** [comment]
@peichman-umd if you can try removing some of the configuration changes, or if you want I can try if you don't have time.

**Comment by peichman-umd** [comment]
@whikloj It looks like most of them are necessary. I tried backing them out, but kept getting 500 errors in the IT because of classes not found and similar errors.

**Pull request merged**